### ISSUE TITLE
python3.9/site-packages/torch/share/cmake': No such file or directory

### ISSUE THREAD

**Issue by takfate** [rocm-arch/python-pytorch-rocm#18]
at the end of building package, the error about lacking `python3.9` occurs. the error logs follow:
```
yte-compiling /home/tak/.cache/yay/python-pytorch-rocm/pkg/python-pytorch-rocm/usr/lib/python3.8/site-packages/caffe2/python/layer_model_helper.py to layer_model_helper.cpython-38.pyc
byte-compiling /home/tak/.cache/yay/python-pytorch-rocm/pkg/python-pytorch-rocm/usr/lib/python3.8/site-packages/caffe2/python/utils_test.py to utils_test.cpython-38.pyc
byte-compiling /home/tak/.cache/yay/python-pytorch-rocm/pkg/python-pytorch-rocm/usr/lib/python3.8/site-packages/caffe2/core/nomnigraph/op_gen.py to op_gen.cpython-38.pyc
byte-compiling /home/tak/.cache/yay/python-pytorch-rocm/pkg/python-pytorch-rocm/usr/lib/python3.8/site-packages/caffe2/core/nomnigraph/__init__.py to __init__.cpython-38.pyc
byte-compiling /home/tak/.cache/yay/python-pytorch-rocm/pkg/python-pytorch-rocm/usr/lib/python3.8/site-packages/caffe2/core/__init__.py to __init__.cpython-38.pyc
writing byte-compilation script '/tmp/tmpr4zghz1d.py'
/usr/bin/python /tmp/tmpr4zghz1d.py
removing /tmp/tmpr4zghz1d.py
running install_egg_info
running egg_info
creating torch.egg-info
writing torch.egg-info/PKG-INFO
writing dependency_links to torch.egg-info/dependency_links.txt
writing entry points to torch.egg-info/entry_points.txt
writing requirements to torch.egg-info/requires.txt
writing top-level names to torch.egg-info/top_level.txt
writing manifest file 'torch.egg-info/SOURCES.txt'
reading manifest file 'torch.egg-info/SOURCES.txt'
writing manifest file 'torch.egg-info/SOURCES.txt'
Copying torch.egg-info to /home/tak/.cache/yay/python-pytorch-rocm/pkg/python-pytorch-rocm/usr/lib/python3.8/site-packages/torch-1.7.0a0-py3.8.egg-info
running install_scripts
Installing convert-caffe2-to-onnx script to /home/tak/.cache/yay/python-pytorch-rocm/pkg/python-pytorch-rocm/usr/bin
Installing convert-onnx-to-caffe2 script to /home/tak/.cache/yay/python-pytorch-rocm/pkg/python-pytorch-rocm/usr/bin
mv: cannot stat '/home/tak/.cache/yay/python-pytorch-rocm/pkg/python-pytorch-rocm/usr/lib/python3.9/site-packages/torch/share/cmake': No such file or directory
==> ERROR: A failure occurred in package_python-pytorch-rocm().
    Aborting...
error making: python-pytorch-rocm (python-pytorch-opt-rocm)

```

**Comment by acxz** [comment]
> Copying torch.egg-info to /home/tak/.cache/yay/python-pytorch-rocm/pkg/python-pytorch-rocm/usr/lib/python3.8/site-packages/torch-1.7.0a0-py3.8.egg-info

hmm for some reason it is installing to a python3.8 installation

**Comment by takfate** [comment]
I directly run bash `yay -S python-pytorch-opt-rocm`. I don't know where is wrong.
This is the all-error log:
[https://gist.github.com/takfate/9dbff0539655c3d03137bdffcdbdb394](url)
Maybe I didn't sync the `AUR` or delete the corresponding caches.
I will try again.

**Comment by acxz** [comment]
> all-error log

it seems that you are using `1.7.1` now, although your log seems to be truncated.

**Comment by takfate** [comment]
Sorry. I seem to copy one wrong link.
The new link follows:
https://gist.github.com/takfate/c39fd4fe5d209d9129d9ef7babe7c9b9

**Comment by acxz** [comment]
I still can't seem to see the error. Do you mind just zooming in on the error message and pasting that?

**Comment by takfate** [comment]
this link must be an entire log!
https://raw.githubusercontent.com/takfate/gist-repo/main/pr.log

**Comment by acxz** [comment]
> -- Found PythonInterp: /usr/bin/python (found suitable version "3.8.5", minimum required is "3.0") 
-- Found PythonLibs: /usr/lib/libpython3.8.so.1.0 (found suitable version "3.8.5", minimum required is "3.0")

Looks like you still have python 3.8 installed.
Can you paste the output of `yay -Q python`?
And then can you run `yay -Syu` and paste the output of `yay -Q python` again?

**Comment by takfate** [comment]
I installed  `python39` by mistake.
Now I have installed successfully `python-pytorch-rocm`.
But I still have some questions:
1. This package `python-pytorch-rocm` does use `git clone pytorch-1.7.1`. Why does this command `Copying torch.egg-info to /home/tak/.cache/yay/python-pytorch-rocm/pkg/python-pytorch-opt-rocm/usr/lib/python3.9/site-packages/torch-1.7.0a0-py3.9.egg-info` still appear at end of installing?
2. When `import torch`. some exceptions were raised:
```
Python 3.9.1 (default, Dec 13 2020, 11:55:53) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.9/site-packages/torch/__init__.py", line 190, in <module>
    from torch._C import *
ImportError: libamdhip64.so.3: cannot open shared object file: No such file or directory
>>> 

```
How should `libamdhip64.so` be installed?

**Comment by acxz** [comment]
> I installed python39 by mistake.

I assume you mean you had installed python38 by mistake.

> 1.

That is weird. Can you run `python -c "import torch; print(torch.__version__)"` to see what that gives you? But you need to make sure `import torch` works first.

> 2.

hmm. `libamdhip64.so` is installed by `hip-rocclr`. Right now `hip-rocclr` installs `libamdhip64.so.4` as `hip-rocclr` is version `4.0.0`. I believe the solution to this is rebuilding `miopen-hip` with an updated `hip-rocclr` version.

Basically try this:
`yay -Rns miopen-hip`
`yay -Syu hip-rocclr`
`yay -Syu aur/miopen-hip`
`yay -Syu python-pytorch-rocm`

Make sure to rebuild packages whenever given the option to do so.

**Comment by takfate** [comment]
1. I have installed `python39` and set a right soft link.When running `python -c "import torch; print(torch.__version__)"`, The exception is same as the previous.
2. I am trying.

**Comment by acxz** [comment]
> I have installed python39 and set a right soft link.When running python -c "import torch; print(torch.__version__)", The exception is same as the previous.

Python 3.9 is the default version of python. There is no need to set a soft link. Can you paste the output of `yay -Q python`? You may have an outdated system which is why you are also seeing error number 2.

Let us resolve this first before moving on.

**Comment by takfate** [comment]
This question may have been no problem.
I have used `yay -Q python` and I make sure that the output of the version of python is 3.9.1.

**Comment by acxz** [comment]
> I had used yay -Q python yesterday and I make sure that the output of the version of python is 3.9.1.

Okay that's good.

>  Now I can't build miopen-hip 4.0.0-2 successfully

Ah totally forgot about that the latest miopen-hip that works is `3.8`. Well then I'm not sure. You should just use the miopen-hip from arch4edu, But that will just recreate the same build env you had in the beginning.

**Comment by takfate** [comment]
So, I should wait for `miopen-hip 4.0.0` to be built successfully.
Is this so?
When will `miopen-hip 4.0.0` of `arch4edu` be available?

**Comment by acxz** [comment]
Whenever someone is kind enough to fix the issue ;)

Although I still believe that the older miopen-hip (3.8) should still work, but I guess that it didn't work in your case.

**Comment by takfate** [comment]
I try again.

**Comment by takfate** [comment]
likely, `import torch` is searching `libamdhip64.so.3` .but i installed `rocclr 4.0.0` with `libamdhip64.so.4`.
maybe i should install `rocclr 3.8.0`?
can `yay` or `pacman` downgrade a specified package?

**Comment by nquest** [comment]
Only rocclr-3.10 available prebuilt in arch4edu repo. miopen-hip is 3.8.

**Comment by acxz** [comment]
I'll go ahead and close this issue due to inactivity. @takfate I'm currently rewriting the PKGBUILD to work with the newer rocm packages and will let you know when you can test it out, to see if the issues you have when importing torch still occur.
